providers/providers.go

```diff
@@ -46,7 +46,9 @@ func New(provider string, p *ProviderData) Provider {
 		return NewNextcloudProvider(p)
 	case "digitalocean":
 		return NewDigitalOceanProvider(p)
-	default:
+	case "google":
 		return NewGoogleProvider(p)
+	default:
+		return nil
```

Comment:
This is a breaking change, we will need to highlight this in the changelog.

Some users may be implicitly using the google provider unknowingly. We need to make a note that the default has changed from `google` to an error and that to retain existing behaviour, if a user hasn't set their provider, to set it to `google`

Reply:
No, it's not, we still default to "google": https://github.com/oauth2-proxy/oauth2-proxy/blob/master/pkg/apis/options/options.go#L137

Reply:
Also this: https://github.com/oauth2-proxy/oauth2-proxy/blob/master/pkg/apis/options/options.go#L219

Reply:
It is still breaking, we just need to make a note.

The default is `google` (which this doesn't change). But the unknown case reverts to `google` -- this will change that.

Gotta warn our users who misspelled `google` as `goggle` 😄

Reply:
Ah, that's what you meant. Will update the doc.

Reply:
Fixed.